pkg/reconciler/instances/connectivityproxy/connectivityproxy.go

```diff
@@ -15,39 +16,44 @@ const (
 	istioConfigPrefix    = "istio"
 )
 
-type CopyFactory func(configs map[string]interface{}, inClusterClientSet, targetClientSet k8s.Interface) *SecretCopy
+type CopyFactory func(task *reconciler.Task, inClusterClientSet, targetClientSet k8s.Interface) *SecretCopy
 
 //nolint:gochecknoinits //usage of init() is intended to register reconciler-instances in centralized registry
 func init() {
 	log := logger.NewLogger(false)
 
 	log.Debugf("Initializing component reconciler '%s'", ReconcilerName)
-	reconciler, err := service.NewComponentReconciler(ReconcilerName)
+	reconcilerInstance, err := service.NewComponentReconciler(ReconcilerName)
 	if err != nil {
 		log.Fatalf("Could not create '%s' component reconciler: %s", ReconcilerName, err)
 	}
 
-	reconciler.
-		WithReconcileAction(&CustomAction{
-			Name:   "action",
-			Loader: &K8sLoader{},
-			Commands: &CommandActions{
-				clientSetFactory: kubeclient.NewInClusterClientSet,
-				targetClientSetFactory: func(context *service.ActionContext) (k8s.Interface, error) {
-					return context.KubeClient.Clientset()
-				},
-				install: service.NewInstall(log),
-				copyFactory: []CopyFactory{
-					registrySecretCopy,
-					istioSecretCopy,
-				},
+	action := CustomAction{
+		Name:   "action",
+		Loader: &K8sLoader{},
+		Commands: &CommandActions{
+			clientSetFactory: kubeclient.NewInClusterClientSet,
+			targetClientSetFactory: func(context *service.ActionContext) (k8s.Interface, error) {
+				return context.KubeClient.Clientset()
 			},
-		})
+			install: service.NewInstall(log),
+			copyFactory: []CopyFactory{
+				registrySecretCopy,
+				istioSecretCopy,
+			},
+		},
+	}
+	reconcilerInstance.
+		WithDeleteAction(&action).
+		WithReconcileAction(&action)
 }
 
-func registrySecretCopy(configs map[string]interface{}, inClusterClientSet, targetClientSet k8s.Interface) *SecretCopy {
+func registrySecretCopy(task *reconciler.Task, inClusterClientSet, targetClientSet k8s.Interface) *SecretCopy {
+	configs := task.Configuration
+	toNamespace := task.Namespace
```

Comment:
Is the `registryConfigPrefix` somehow preserved or it should be dropped there?

Reply:
It should be preserved.

Reply:
Good! :D